mptt/admin.py

```diff
@@ -92,28 +94,44 @@ def get_actions(self, request):
         return actions
 
 
+class JS(object):
+    """
+    Use this to insert a script tag via ``forms.Media`` containing additional
+    attributes (such as ``id`` and ``data-*`` for CSP-compatible data
+    injection.)
+    """
+    def __init__(self, js, attrs):
+        self.js = js
+        self.attrs = attrs
+
+    def startswith(self, _):
+        # Masquerade as absolute path so that we are returned as-is.
+        return True
+
+    def __html__(self):
+        return format_html(
```

Comment:
Of course! 9dafd7686c729bf36b49f22231ea7cd55bc01ba3

This hack should work really well also with prehistoric versions of Django, back to https://github.com/django/django/commit/a19ed8aea395e8e07164ff7d85bd7dff2f24edca (the newforms-admin merge). The behavior of returning `TemplateResponse` instances from `changelist_view` was introduced with https://github.com/django/django/commit/c8092b840b4151724a6babdbeb84cf3138057c6a in 2011. 

(Adding data attributes to script tags is not my own idea btw, I got it here: https://github.com/django/django/blob/master/django/contrib/admin/templates/admin/popup_response.html)